src/extension.ts

```diff
@@ -57,13 +101,39 @@ async function fetchPackageMetadata(requirement: ProjectNameRequirement): Promis
     return await metadataCache.get(requirement.name)!()
 }
 
+let lastLine = -1
+let lastHover: vscode.Hover | null = null
+let lastLineText = ''
```

Comment:
I think it'd be better to make this state part of the hover provider class instead of global vars (I'm now more allergic to global vars than when I wrote this extension originally). I'm also unsure if `lastLine` actually matters, I'd say `lastLineText` is enough of a cache key